docs/staking-entry-queue/dd.md

```diff
@@ -0,0 +1,136 @@
+# [Project Name] Design Document
+
+- Owner: @just-mitch
+- Approvers:
+  - @LHerskind
+  - @Maddiaa0
+  - @aminsammara
+- Target DD Approval Date: 2025-06-10
+- Target Project Delivery Date: 2025-06-15
+
+## Executive Summary
+
+Requirement: Mitigate the risk of a large influx of byzantine stakers which could threaten the stability/security of the network.
+
+We accomplish this by introducing a staking entry queue.
+
+That is, when a staker deposits into the aztec network, they are added to a queue on L1.
+
+## Timeline
+
+Outline the timeline for the project. E.g.
+
+- Build L1 queue: 1 day
+- Update typescript: 1 day
+- Review/polish: 1 day
+
+Total: 3 days
+
+## Introduction
+
+It is not good for it to be possible for a large number of validators to (near) instantly start participating in block production.
+
+This can result in, for example, where a significant portion of the validator set is malicious or faulty.
+
+This is a particular concern at the outset of a network where it is possible for a large influx to occur in the absence of any flow control mechanism.
+
+This can be problematic for the following reasons:
+
+- It is possible that the nodes are improperly configured (accidentally or otherwise), and are not actually able to participate in block production
+- It is possible that the p2p network will not form a well connected mesh, hindering throughput
+
+It is better to allow a more gradual transition, rather than shock the system.
+
+## Interface
+
+Users are the would-be and current validators of the aztec network.
+
+Currently, users are able to go directly to rollup instances, and call
+
+```solidity
+  function deposit(address _attester, address _withdrawer, bool _onCanonical) external;
+```
+
+This interface will remain _unchanged_, but the semantics will be that they are now "enqueued".
+
+Users _will_ need to call `flushEntryQueue` though to move into the active state of validating.
+
+```solidity
+  function flushEntryQueue() external;
+```
+
+## Implementation
+
+When would-be validators call `deposit`, they will enter a queue maintained by the rollup instance.
+
+Validators in the queue do not participate in block production, and will not be sampled during committee selection.
+
+When _anyone_ calls `flushEntryQueue` for a given epoch `e`, then up to `N` validators will be dequeued and added to the validator set, and be eligible to serve on the committee for epoch `e`.
```

Comment:
I think the notion of "being able to ser ve on `e` is not good here. For the simple reason that it depends on the validator sampling, so if the "delay" in there changes I don't see why it should change much in the queues. It is fine that it is only once per epoch so there is some time frame where it can happen, but would not say anything about the time where they are eligible as that is really depending on the other parts, mainly that when queue is flushed they get added to the validator set